### ISSUE TITLE
python version laster then 3.12， canvas@2.9.1 will install fail,  because distutils removed.

### ISSUE THREAD

**Issue by zhaxie** [apitable/apitable#1746]
If your python version laster then 3.12， canvas@2.9.1 will install fail。    you can install python@3.10 resolved it。  
[canvas-install.txt](https://github.com/apitable/apitable/files/15349894/canvas-install.txt)

**Comment by Jeffrey-mu** [comment]
@zhaxie 
![image](https://github.com/apitable/apitable/assets/66287770/1773ce9b-a3a2-4bfb-90a5-48382c74df16)
Running Nest Error。

**Comment by mr-kelly** [comment]
You are absolutely right. `node-gyp` under Node.js 16/18 relies on Python's `distutils`, which was removed in Python 3.12. Using Python 3.10 or 3.11 is the recommended workaround. Thank you for providing the solution to the community!

---

您说得非常对。在 Node.js 16/18 环境下的 `node-gyp` 依赖 Python 的 `distutils`，而该模块在 Python 3.12 中已被移除。降级使用 Python 3.10 或 3.11 是正确的解决办法。感谢您为社区提供这个解决方案！
